src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java

```diff
@@ -0,0 +1,179 @@
+            final Instant start = Instant.now();
+            HttpHeaders responseHeaders = restTemplate.exchange(applicationParams.draftBaseURL(),
+                                                                HttpMethod.POST,
+                                                                requestEntity,
+                                                                HttpEntity.class).getHeaders();
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Create", duration.toMillis(), true);
+            return getDraftId(responseHeaders);
+        } catch (Exception e) {
+            LOG.warn("Error while saving draft", e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public DraftResponse update(final UpdateCaseDraftRequest draft, final String draftId) {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(draft, headers);
+            final Instant start = Instant.now();
+            restTemplate.exchange(applicationParams.draftURL(draftId), HttpMethod.PUT, requestEntity, HttpEntity.class);
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Update", duration.toMillis(), true);
+            return aDraftResponse()
+                .withId(draftId)
+                .build();
+        } catch (HttpClientErrorException e) {
+            LOG.warn("Error while updating draftId={}", draftId, e);
+            if (e.getRawStatusCode() == RESOURCE_NOT_FOUND) {
+                throw new ResourceNotFoundException(String.format(RESOURCE_NOT_FOUND_MSG, draftId));
+            }
+        } catch (Exception e) {
+            LOG.warn("Error while updating draftId={}", draftId, e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
+        }
+        return null;
+    }
+
+    @Override
+    public DraftResponse get(final String draftId) {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(headers);
+            final Instant start = Instant.now();
+            Draft draft = restTemplate.exchange(applicationParams.draftURL(draftId), HttpMethod.GET, requestEntity, Draft.class).getBody();
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Get", duration.toMillis(), true);
+            return assembleDraft(draft);
+        } catch (HttpClientErrorException e) {
+            LOG.warn("Error while getting draftId=" + draftId, e);
```

Comment:
can we use placeholders rather than string concatenation?

Reply:
Done.